augur/refine.py

```diff
@@ -241,11 +241,27 @@ def run(args):
     if args.divergence_units=='mutations-per-site': #default
         pass
     elif args.divergence_units=='mutations':
-        L = tt.seq_len
-        for node in node_data['nodes']:
+        if not args.timetree:
+            tt.infer_ancestral_sequences()
+        nuc_map = profile_maps['nuc']
+
+        def different(a,d):
+            '''
+            determine whether two ancestral states should count as mutation for divergence estimates
+            while correctly accounting for ambiguous nucleotides
+            '''
+            if a in ['-', 'N'] or d in ['-', 'N']:
+                return False
+            elif a in nuc_map and d in nuc_map:
+                return np.sum(nuc_map[a]*nuc_map[d])==0
```

Comment:
I didn't understand this logic until I looked up the definition of the profile maps in TreeTime's source code. This function is a generally useful utility, but it would make more sense living in TreeTime's `seq_utils.py` next to the profile map definitions themselves. This would allow augur to use the function without needing to know the internals of how profile maps work in TreeTime.

Reply:
yes, I could add this. But we could move it at a later point.